addons/binding/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/internal/model/Realtime.java

```diff
@@ -18,35 +16,42 @@
  */
 public class Realtime extends ErrorResponse {
 
-    @SerializedName(value = "current", alternate = "current_ma")
+    private static final int MILLIWATT_TO_WATT = 1000;
+    private static final int MILLIAMP_TO_AMP = 1000;
+    private static final int WATTHOUR_TO_KILOWATTHOUR = 1000;
+    private static final int MILLIVOLT_TO_VOLT = 1000;
+
     private double current;
-    @SerializedName(value = "power", alternate = "power_mw")
     private double power;
-    @SerializedName(value = "total", alternate = "total_wh")
     private double total;
-    @SerializedName(value = "voltage", alternate = "voltage_mv")
     private double voltage;
 
+    // JSON names used for v2 hardware
+    private double currentMa;
```

Comment:
I think that if you would make it a `Double` (all of them) you could just check for null and imho a helper method would make the code better:

```java
return calculateIfNeeded(currentMa , curreny, MILLIAMP_TO_AMP);

double calculateIfNeeded(Double newValue, Double fallback, int factor) {
   if (newValue == null) {
       return fallback;
   } else {
      return newValue / factor;
   }
} 
```

Reply:
Technically you would need to check `fallback` for null values too because if it's null you will get a nasty exception... There should be a line when optimization is enough and I'm fine with the current (:smile:) code.

Reply:
My personal main reason is if we use `double` a missing value is converted to `0.0`, while the default of `Double``null` would be a clearer way to represent a missing value instead of using  the arbitrary value `0.0` which can also be there for other reasons.

Reply:
@martinvw, I have a commit ready with the suggested `calculatedIfNeeded` method so will push if that's how you'd like to proceed.  Though I feel that @Hilbrand's approach does read a bit more gracefully.   

```java

    public double getVoltage() {
        return calculateIfNeeded(voltageMv, voltage, MILLIVOLT_TO_VOLT);
    }

    private double calculateIfNeeded(Double newValue, Double fallback, int factor) {
        if (newValue == null) {
            if (fallback == null)
                return 0.0;

            return fallback;
        } else {
            return newValue / factor;
        }
    }
```
Perhaps the helper class might read better as `refactorIfNeeded`?

Reply:
@martinvw In this case I don't think it doesn't add much to the solution and is a matter of personal preference. Unless the method would return a `Double` it would make sense and even then it needs more additional work for a use case that is unknown to exist in the field. But should we spend so much time on such a little preference? There are far more bigger issues to tackle so I think we should better spend our time to other issues than this minor improvement. Details are important, but we should not go to far imho. 
@kcstonacek the `fallback` variable can be a `double` because on null check it falls back to 0. This would eliminate the need for the null check. But then again I don't think it should be changed at all.

Reply:
> But should we spend so much time on such a little preference?

No but I assumed it to be a simple and quick fix and still see it as not a good design to use a `double` `0.0` to represent either the value 0.0 and/or the fact that it is missing.